### PR TITLE
add solution: MovieController and MovieSessionController

### PR DESCRIPTION
**Pull request by Victoria-Moiseienko**

### CODE REVIEW

File: src/main/java/vic/cinema/controller/AuthenticationController.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+import vic.cinema.dto.UserRequestDto;
+import vic.cinema.security.AuthenticationService;
+
+@RestController
+public class AuthenticationController {
+    //Register - POST: /register
```
Comment (dmytro-hryhoruk) [inline src/main/java/vic/cinema/controller/AuthenticationController.java]: Do you need this comment?

File: src/main/java/vic/cinema/controller/MovieSessionController.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+    @PostMapping
+    public MovieSession addMovieSession(
+            @RequestBody MovieSessionRequestDto movieSessionRequestDto) {
+        MovieSession movieSession =
+                movieSessionService.add(movieSessionMapper.getMovieSession(movieSessionRequestDto));
+        return movieSession;
+    }
+
+    @GetMapping
```
Comment (dmytro-hryhoruk) [inline src/main/java/vic/cinema/controller/MovieSessionController.java]:
```suggestion
    @GetMapping("/available")
```

File: src/main/java/vic/cinema/dao/CinemaHallDao.java [hunk 3]
```diff
@@ -7,4 +7,6 @@
     CinemaHall add(CinemaHall cinemaHall);
 
     List<CinemaHall> getAll();
+
+    CinemaHall getById(Long id);
```
Comment (dmytro-hryhoruk) [inline src/main/java/vic/cinema/dao/CinemaHallDao.java]:
```suggestion
    CinemaHall get(Long id);
```

File: src/main/java/vic/cinema/dao/impl/CinemaHallDaoImpl.java [hunk 4]
```diff
@@ -53,4 +53,16 @@ public CinemaHall add(CinemaHall cinemaHall) {
...
         }
     }
+
+    @Override
+    public CinemaHall getById(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            Query<CinemaHall> query = session.createQuery(
+                    "from CinemaHall where id = :cinemaHallId", CinemaHall.class);
+            query.setParameter("cinemaHallId", id);
+            return query.getSingleResult();
```
Comment (dmytro-hryhoruk) [inline src/main/java/vic/cinema/dao/impl/CinemaHallDaoImpl.java]: https://mate-academy.github.io/jv-program-fulltime/04_spring/checklist/hw_33.html

File: src/main/java/vic/cinema/dao/impl/MovieDaoImpl.java [hunk 5]
```diff
@@ -52,4 +52,15 @@ public Movie add(Movie movie) {
             throw new DataProcessingException("Can't get all movies", e);
         }
     }
+
+    @Override
+    public Movie getById(Long id) {
+        try (Session session = sessionFactory.openSession()) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/vic/cinema/dao/impl/MovieDaoImpl.java:58]: same

File: src/main/java/vic/cinema/dao/impl/MovieDaoImpl.java [hunk 6]
```diff
@@ -52,4 +52,15 @@ public Movie add(Movie movie) {
             throw new DataProcessingException("Can't get all movies", e);
         }
     }
+
+    @Override
+    public Movie getById(Long id) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/vic/cinema/dao/impl/MovieDaoImpl.java]:
```suggestion
    public Movie get(Long id) {
```

File: src/main/java/vic/cinema/dto/MovieSessionDto.java [hunk 7]
```diff
@@ -0,0 +1,4 @@
+package vic.cinema.dto;
+
+public class MovieSessionDto {
```
Comment (dmytro-hryhoruk) [inline src/main/java/vic/cinema/dto/MovieSessionDto.java]: ?

File: src/main/java/vic/cinema/mapper/MovieMapper.java [hunk 8]
```diff
@@ -0,0 +1,24 @@
+package vic.cinema.mapper;
+
+import org.springframework.stereotype.Component;
+import vic.cinema.dto.MovieRequestDto;
+import vic.cinema.dto.MovieResponseDto;
+import vic.cinema.model.Movie;
+
+@Component
+public class MovieMapper {
+    public MovieResponseDto getMovieResponseDto(Movie movie) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/vic/cinema/mapper/MovieMapper.java]:
```suggestion
    public MovieResponseDto toDto(Movie movie) {
```

File: src/main/java/vic/cinema/mapper/MovieMapper.java [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+        MovieResponseDto responseDto = new MovieResponseDto();
+        responseDto.setMovieId(movie.getId());
+        responseDto.setDescription(movie.getDescription());
+        responseDto.setTitle(movie.getTitle());
+        return responseDto;
+    }
+
+    public Movie getMovie(MovieRequestDto movieRequestDto) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/vic/cinema/mapper/MovieMapper.java]:
```suggestion
    public Movie toMovie(MovieRequestDto movieRequestDto) {
```

**Review by dmytro-hryhoruk — COMMENTED** [review]
apply comments about naming to all suitable places

File: src/main/java/vic/cinema/controller/AuthenticationController.java (hunk 1 above)
Comment (Victoria-Moiseienko) [reply src/main/java/vic/cinema/controller/AuthenticationController.java]: Done.

File: src/main/java/vic/cinema/controller/MovieSessionController.java (hunk 2 above)
Comment (Victoria-Moiseienko) [reply src/main/java/vic/cinema/controller/MovieSessionController.java]: Done

File: src/main/java/vic/cinema/dao/CinemaHallDao.java (hunk 3 above)
Comment (Victoria-Moiseienko) [reply src/main/java/vic/cinema/dao/CinemaHallDao.java]: Done

File: src/main/java/vic/cinema/dao/impl/MovieDaoImpl.java (hunk 6 above)
Comment (Victoria-Moiseienko) [reply src/main/java/vic/cinema/dao/impl/MovieDaoImpl.java]: Done

File: src/main/java/vic/cinema/dto/MovieSessionDto.java (hunk 7 above)
Comment (Victoria-Moiseienko) [reply src/main/java/vic/cinema/dto/MovieSessionDto.java]: Done

File: src/main/java/vic/cinema/mapper/MovieMapper.java (hunk 8 above)
Comment (Victoria-Moiseienko) [reply src/main/java/vic/cinema/mapper/MovieMapper.java]: Done

File: src/main/java/vic/cinema/mapper/MovieMapper.java (hunk 9 above)
Comment (Victoria-Moiseienko) [reply src/main/java/vic/cinema/mapper/MovieMapper.java]: Done

File: src/main/java/vic/cinema/dao/impl/MovieDaoImpl.java (hunk 5 above)
Comment (Victoria-Moiseienko) [reply src/main/java/vic/cinema/dao/impl/MovieDaoImpl.java:58]: Renamed.

File: src/main/java/vic/cinema/dao/impl/CinemaHallDaoImpl.java (hunk 4 above)
Comment (Victoria-Moiseienko) [reply src/main/java/vic/cinema/dao/impl/CinemaHallDaoImpl.java]: Updated

File: src/main/java/vic/cinema/controller/MovieController.java [hunk 10]
```diff
@@ -26,15 +26,13 @@ public MovieController(MovieService movieService, MovieMapper movieMapper) {
 
     @PostMapping
     public Movie addMovie(@RequestBody MovieRequestDto movieRequestDto) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/vic/cinema/controller/MovieController.java]: let's make this method void and later we may return http status

**Review by dmytro-hryhoruk — APPROVED** [review]

File: src/main/java/vic/cinema/controller/MovieController.java (hunk 10 above)
Comment (Victoria-Moiseienko) [reply src/main/java/vic/cinema/controller/MovieController.java]: Done

**Pull request merged**